docs/CACHING.md

```diff
@@ -0,0 +1,353 @@
+# Caching Configuration of Mountpoint for Amazon S3 CSI Driver
```

Comment:
We already have `FEATURES.md` file [here](https://github.com/awslabs/mountpoint-s3-csi-driver/blob/v2/docs/FEATURES.md) and caching seems like a feature. Should we move it into that file or extract Pod Sharing into separate `MOUNTPOINT_POD_SHARING.md` doc? I think I prefer the latter.